code/game/objects/items/pinpointer.dm

```diff
@@ -97,16 +97,14 @@
 
 /obj/item/pinpointer/crew/proc/trackable(mob/living/carbon/human/H)
 	var/turf/here = get_turf(src)
-	if((H.z == 0 || H.z == here.z) && istype(H.w_uniform, /obj/item/clothing/under))
-		var/obj/item/clothing/under/U = H.w_uniform
-
-		// Suit sensors must be on maximum.
-		if(!U.has_sensor || (U.sensor_mode < SENSOR_COORDS && !ignore_suit_sensor_level))
-			return FALSE
-
-		var/turf/there = get_turf(H)
-		return (H.z != 0 || (there && there.z == here.z))
-
+	var/turf/there = get_turf(H)
+	if((H.z == 0 || H.z == here.z) && (H.z != 0 || (there && there.z == here.z)))
```

Comment:
The existing z-logic seems kind of tortured, I'm not sure if this suggestion is equivalent, but it seems much much more simpler. Does this make sense?
```suggestion
	if(here && there && here.z == there.z && here.z != 0)
```

Reply:
To be honest, I don't understand what Z 0 is and why is it checked, so I just moved the statement without changing it. I assume that the target should be on the same Z as the device (and not Z0?..) - Then your statement is sure better.

`/datum/computer_file/program/radar/proc/trackable(atom/movable/signal)` doesn't check Z0, it checks ZTRAIT_STATION instead. We probably should make these consistent then.

Reply:
I assume that I should use trait to make things work on icebox property? Marked PR as draft for research.